src/host.cpp

```diff
@@ -41,4 +35,34 @@ void remove_host(CopyOnWriteHostVec& hosts, const SharedRefPtr<Host>& host) {
   }
 }
 
+void Host::LatencyTracker::update(uint64_t latency_ns) {
+  uint64_t now = uv_hrtime();
+
+  ScopedSpinlock l(SpinlockPool<LatencyTracker>::get_spinlock(this));
+
+  TimestampedAverage previous = current_;
+
+  if (previous.num_measured < threshold_to_account_) {
+    current_.average = -1;
+    current_.num_measured = previous.num_measured + 1;
+    current_.timestamp = now;
```

Comment:
num_measured and timestamp update are identical for all branches. Could combine.